packages/zoe/src/contracts/coveredCall.js

```diff
@@ -1,84 +1,94 @@
- * valuable good.
+ *
+ * This deadline serves as the expiration date for the covered call
+ * option. After this deadline, if the option has not been exercised,
+ * the underlying assets are automatically paid out to the creator of
+ * the contract as a refund.
+ *
+ * After the owner of the digital assets escrows the assets in the
+ * initial offer, they receive a seat. The payout for this seat will
+ * either be a refund of the underlying assets (as mentioned above) or
+ * the strike price. Zoe's enforcement of offer safety guarantees that
+ * the payout is either a refund or the strike price, regardless of
+ * whether this contract is buggy.
+ *
+ * The offerResult of this initial seat resolves to the call option
+ * itself: an assayable invitation to buy the underlying assets. The
+ * call option invitation has this additional information in the
+ * value: { expirationDate, timerAuthority, underlyingAssets,
+ * strikePrice }
+ *
+ * The invitation itself can be traded as a valuable digital asset: a
+ * covered call option.
+ *
+ * The recipient of a covered call option (buying it on some other
+ * exchange or through some other trading contract) can exercise the
+ * option before the deadline by using the option as an invitation to
+ * this contract, paying the strike price and receiving the underlying
+ * assets. The recipient of a covered call option can use whatever
+ * keywords they wish, as long as they give the strike price as
+ * specified in the invitation value, and want the underlying assets
+ * exactly.
  *
  * @type {ContractStartFn}
  */
 const start = zcf => {
-  assertIssuerKeywords(zcf, harden(['UnderlyingAsset', 'StrikePrice']));
+  const sellSeatExpiredMsg = `The covered call option is expired.`;
 
-  const makeCallOption = sellerSeat => {
-    assertProposalShape(sellerSeat, {
-      give: { UnderlyingAsset: null },
-      want: { StrikePrice: null },
-      exit: { afterDeadline: null },
-    });
-    const { want, give, exit } = sellerSeat.getProposal();
-    const rejectMsg = `The covered call option is expired.`;
+  const makeOption = sellSeat => {
+    assertProposalShape(sellSeat, { exit: { afterDeadline: null } });
 
-    const exerciseOption = exerciserSeat => {
-      assertProposalShape(exerciserSeat, {
-        give: { StrikePrice: null },
-        want: { UnderlyingAsset: null },
-      });
-      const swapResult = swap(zcf, sellerSeat, exerciserSeat, rejectMsg);
+    const exerciseOption = buySeat => {
+      try {
+        swapExact(zcf, sellSeat, buySeat, sellSeatExpiredMsg);
+      } catch (err) {
+        console.log(
+          'Swap failed. Please make sure your offer has the same underlyingAssets and strikePrice as specified in the invitation details. The keywords should not matter.',
+        );
+        throw err;
+      }
       zcf.shutdown();
-      return swapResult;
+      return `The option was exercised. Please check your payout to receive the underlying assets.`;
```

Comment:
I don't think "Please check your payout" is the right terminology. Maybe "Please use" or "Please exercise"? "check" isn't active enough.

Reply:
Not `exercise`. Especially not in a call options contract! Perhaps `Please collect the asset in your payout.` ?

Reply:
Done